### PR TITLE
refactor(search) Use search across multiple-entities API, deprecate Aggregator classes

### PR DESCRIPTION
**Pull request by iprentic**
* Added tests to ensure parity with the old functionality w.r.t the legacy entity aggregation group

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-io/src/main/java/com/linkedin/metadata/search/SearchService.java [hunk 1]
```diff
@@ -95,7 +100,49 @@ public SearchResult searchAcrossEntities(@Nonnull List<String> entities, @Nonnul
...
+      facets = new ArrayList<>(facets);
+      facets.add(INDEX_VIRTUAL_FIELD);
+    }
+    List<String> nonEmptyEntities;
+    List<String> lowercaseEntities = entities.stream().map(String::toLowerCase).collect(Collectors.toList());
+    try (Timer.Context ignored = MetricUtils.timer(this.getClass(), "getNonEmptyEntities").time()) {
+      nonEmptyEntities = _entityDocCountCache.getNonEmptyEntities();
+    }
+    if (!entities.isEmpty()) {
+      nonEmptyEntities = nonEmptyEntities.stream().filter(lowercaseEntities::contains).collect(Collectors.toList());
```
Comment (RyanHolstien) [inline metadata-io/src/main/java/com/linkedin/metadata/search/SearchService.java:118]:
```suggestion
      entities = nonEmptyEntities.stream().filter(lowercaseEntities::contains).collect(Collectors.toList());
```

File: metadata-io/src/main/java/com/linkedin/metadata/search/SearchService.java [hunk 2]
```diff
@@ -95,7 +100,49 @@ public SearchResult searchAcrossEntities(@Nonnull List<String> entities, @Nonnul
...
+    }
+    SearchResult result = _cachingEntitySearchService.search(nonEmptyEntities, input, postFilters, sortCriterion, from, size, searchFlags, facets);
```
Comment (RyanHolstien) [inline metadata-io/src/main/java/com/linkedin/metadata/search/SearchService.java:120]:
```suggestion
    SearchResult result = _cachingEntitySearchService.search(entities, input, postFilters, sortCriterion, from, size, searchFlags, facets);
```

File: metadata-io/src/main/java/com/linkedin/metadata/search/elasticsearch/query/request/SearchFieldConfig.java [hunk 3]
```diff
@@ -148,6 +150,27 @@ private static String getAnalyzer(String fieldName, SearchableAnnotation.FieldTy
         }
     }
 
+    @Override
```
Comment (RyanHolstien) [inline metadata-io/src/main/java/com/linkedin/metadata/search/elasticsearch/query/request/SearchFieldConfig.java:153]: You can use lombok for these as well

**Review by RyanHolstien — APPROVED** [review]

File: metadata-io/src/main/java/com/linkedin/metadata/search/elasticsearch/query/request/SearchFieldConfig.java (hunk 3 above)
Comment (iprentic) [reply metadata-io/src/main/java/com/linkedin/metadata/search/elasticsearch/query/request/SearchFieldConfig.java]: Nice, done

**Pull request merged**